sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -630,8 +639,18 @@ object AdaptiveSparkPlanExec {
   /**
    * Apply a list of physical operator rules on a [[SparkPlan]].
    */
-  def applyPhysicalRules(plan: SparkPlan, rules: Seq[Rule[SparkPlan]]): SparkPlan = {
-    rules.foldLeft(plan) { case (sp, rule) => rule.apply(sp) }
+  def applyPhysicalRules(
+      batchName: String,
+      plan: SparkPlan,
+      rules: Seq[Rule[SparkPlan]]): SparkPlan = {
+    val planChangeLogger = new PlanChangeLogger[SparkPlan]()
```

Comment:
shall we create `PlanChangeLogger` instance only once? (keep it as a class member)